Modules/_posixsubprocess.c

```diff
@@ -936,8 +936,31 @@ subprocess_fork_exec(PyObject *module, PyObject *args)
 #ifdef VFORK_USABLE
     /* Use vfork() only if it's safe. See the comment above child_exec(). */
     sigset_t old_sigs;
-    if (preexec_fn == Py_None &&
-        !call_setuid && !call_setgid && !call_setgroups) {
+    int allow_vfork;
+    if (preexec_fn == Py_None) {
```

Comment:
It seems strange to leave `!call_setuid && !call_setgid && !call_setgroups` out from this check: why bother checking the knob if we're not going to use vfork() anyway? Not that it really matters :)

Reply:
I generated optimized the look of the code for this 3.10 version for the simplest looking diff. the compiler is going to generate effectively the same code in an opt build regardless and this is never the slow part anyways :)

Reply:
_(my initial opening commit in the PR had it structured as you might seem, it made the diffs less obvious as to the actual change)_